From 758207db2a489415f00c4241cd08897d47a9c4ae Mon Sep 17 00:00:00 2001
From: Francois Daoust <fd@tidoust.net>
Date: Thu, 10 Jul 2025 11:54:01 +0200
Subject: [PATCH] Drop pointer-events dfn, adjust cursor-image syntax

The definition of `pointer-events` is not yet as complete as that in the SVG
spec. Tracking issue at:
https://github.com/w3c/csswg-drafts/issues/4438

Syntax of `<cursor-image>` is valid but stacking of `{2}` and `?` is not
supported by the CSSTree parser. Tracking issue at:
https://github.com/csstree/csstree/issues/346

(Why the spec defines the type both as a `value` and a `type`, I do not know,
but this should not have any impact though, so left for another day...)
---
 ed/css/css-ui.json | 36 ++----------------------------------
 1 file changed, 2 insertions(+), 34 deletions(-)

diff --git a/ed/css/css-ui.json b/ed/css/css-ui.json
index e65d3cfd00..1aa20eb8b1 100644
--- a/ed/css/css-ui.json
+++ b/ed/css/css-ui.json
@@ -121,7 +121,7 @@
       "values": [
         {
           "name": "<cursor-image>",
-          "value": "[ <url> | <url-set> ] <number>{2}?",
+          "value": "[ <url> | <url-set> ] [<number>{2}]?",
           "href": "https://drafts.csswg.org/css-ui-4/#valdef-cursor-cursor-image",
           "type": "value"
         },
@@ -129,7 +129,7 @@
           "name": "<cursor-image>",
           "href": "https://drafts.csswg.org/css-ui-4/#typedef-cursor-cursor-image",
           "type": "type",
-          "value": "[ <url> | <url-set> ] <number>{2}?"
+          "value": "[ <url> | <url-set> ] [<number>{2}]?"
         },
         {
           "name": "<url-set>",
@@ -627,38 +627,6 @@
         "userSelect"
       ]
     },
-    {
-      "name": "pointer-events",
-      "href": "https://drafts.csswg.org/css-ui-4/#propdef-pointer-events",
-      "value": "auto | none",
-      "initial": "auto",
-      "appliesTo": "all elements",
-      "inherited": "yes",
-      "percentages": "N/A",
-      "computedValue": "specified keyword",
-      "canonicalOrder": "per grammar",
-      "animationType": "by computed value type",
-      "values": [
-        {
-          "name": "auto",
-          "prose": "Hit-testing occurs normally.",
-          "href": "https://drafts.csswg.org/css-ui-4/#valdef-pointer-events-auto",
-          "type": "value",
-          "value": "auto"
-        },
-        {
-          "name": "none",
-          "prose": "Hit-testing acts as if the boxes generated by the element (see [CSS-DISPLAY-3]) were not there, effectively causing the element behind the pointer-events: none element to become the target of hit-testing-based events instead. The statement above needs to be qualified; for some purposes, hit-testing continues to take the element into account. For instance, clicking and dragging to start a text selection will continue to work as usual. What is the precise set of things for which hit-testing is affected by pointer-events?",
-          "href": "https://drafts.csswg.org/css-ui-4/#valdef-pointer-events-none",
-          "type": "value",
-          "value": "none"
-        }
-      ],
-      "styleDeclaration": [
-        "pointer-events",
-        "pointerEvents"
-      ]
-    },
     {
       "name": "interactivity",
       "href": "https://drafts.csswg.org/css-ui-4/#propdef-interactivity",
-- 
2.37.1.windows.1

